pkg/yurtctl/util/edgenode/util.go

```diff
@@ -0,0 +1,105 @@
+package edgenode
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strings"
+)
+
+func FileExists(filename string) (bool, error) {
+	if _, err := os.Stat(filename); os.IsExist(err) {
+		return true, err
+	} else if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+// GetContentFormFile returns all strings that match the regular expression regularExpression
+func GetContentFormFile(filename string, regularExpression string) ([]string, error) {
+	content, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	ct := string(content)
+	reg := regexp.MustCompile(regularExpression)
+	res := reg.FindAllString(ct, -1)
+	return res, nil
+}
+
+func GetSingleContentFromFile(filename string, regularExpression string) (string, error) {
+	contents, err := GetContentFormFile(filename, regularExpression)
+	if err != nil {
+		return "", fmt.Errorf("failed to read file %s, %v", filename, err)
+	}
+	if contents == nil {
+		return "", fmt.Errorf("no matching string %s in file %s", regularExpression, filename)
+	} else if len(contents) > 1 {
+		return "", fmt.Errorf("there are multiple matching string %s in file %s", regularExpression, filename)
+	}
+	return contents[0], nil
+}
+
+func DirExists(dirname string) (bool, error) {
+	s, err := os.Stat(dirname)
+	if err != nil {
+		return false, err
+	}
+	if !s.IsDir() {
+		return false, fmt.Errorf("%s is not a dir", dirname)
+	}
+	return true, nil
+}
+
+func CopyFile(sourceFile string, destinationFile string) error {
+	content, err := ioutil.ReadFile(sourceFile)
+	if err != nil {
+		return fmt.Errorf("failed to read source file %s: %v", sourceFile, err)
+	}
+	err = ioutil.WriteFile(destinationFile, content, 0666)
+	if err != nil {
+		return fmt.Errorf("failed to write destination file %s: %v", destinationFile, err)
+	}
+	return nil
+}
+
+func ReplaceRegularExpression(content string, replace map[string]string) string {
+	for old, new := range replace {
+		reg := regexp.MustCompile(old)
+		content = reg.ReplaceAllString(content, new)
+	}
+	return content
+}
+
+func GetNodeName() (string, error) {
```

Comment:
how about  use env NODE_NAME in convert and revert command instead of new GetNodeName() function?

Reply:
Does NODE_NAME refer to the env HOSTNAME? 
If so, kubelet startup parameters `--hostname-override` can be used to configure the host name of the node displayed in the cluster,  so the node name in cluster can be different from env HOSTNAME. And it need to get the node name in cluster accurately, because we need to get node resources through node name later.
So maybe it's better to use GetNodeName() function?

Reply:
using NODE_NAME means we can add the following setting to the pod spec
```yaml
...
env:
- name: NODE_NAME
  valueFrom:
    fieldRef:
      fieldPath: spec.nodeName
...
```
But yes, I think we should try to get the correct nodename from the service configuration directly, if `--hostname-override` can be used to overwrite the nodename.

Reply:
@Peeknut as @charleszheng44 had said, env NODE_NAME is set in ServantJobTemplate by `spec.nodeName` field, so in spite of `--hostname-override` is set or not, we can get the node name accurately by env NODE_NAME .

Reply:
That's right. For `yurtctl convert/revert`, we can directly use NODE_NAME. However, for directly calling `yurtctl convert/revert edgenode`, NODE_NAME may not be set, we need to read the configuration file to get it. Both approaches may be required. Maybe I should add steps to read env NODE_NAME in GetNodeName() function?